apps/fulfillment-e2e/src/integration/login.cy.ts

```diff
@@ -1,11 +1,29 @@
 import { PickingListsFragment } from '../support/page_fragments/picking-lists.fragment';
+import { LoginPage } from '../support/page_objects/login.page';
 
 const pickingListsFragment = new PickingListsFragment();
+const loginPage = new LoginPage();
 
 describe('Login Suite', () => {
-  it('should login successfully', () => {
-    cy.login();
+  describe('when logged out', () => {
+    it('should redirect to login page', () => {
+      cy.visit('/');
 
-    pickingListsFragment.getWrapper().should('be.visible');
+      cy.location('pathname').should('to.match', /^\/login/);
+    });
+  });
+  describe('when logging in', () => {
+    it('should login successfully', () => {
+      let isLoadingPage = false;
+
+      cy.on('url:changed', (newurl) => {
+        isLoadingPage = newurl.includes('oauth') ? true : isLoadingPage;
+      });
+
+      cy.login();
+      cy.then(() => expect(isLoadingPage).to.be.true);
+      pickingListsFragment.getPickingListsItem().should('have.length', 2);
```

Comment:
I also wonder about this check -> will we always have 2 pickingListItems, or it might change unexpectedly?

Reply:
One of the ticket requirements was to check for the number of picking list items. Is there a better way to get this number in e2e tests?

Reply:
We can change the requirements to make them more stable :) 

We should check that the correct page (picking list page) is opened, and to prove that we can do 2 checks:
1. wrapper check (which we have on line 26)
2. URL check `cy.url().should('contain', '/picking-list')` or something similar

In that case we don't care about the number of products displayed (because it can be 0, 1, 2, 100500 - so flaky) but still we are able to prove that login logic and redirect works (and that's what we need here)